comments/loading_perf.md

```diff
@@ -0,0 +1,114 @@
+# Reduce map loading times
+
+Page load can take more than ten seconds for maps that have a large amount of model data. Per [Website Response Times] we should have _something_ rendered in under a second, and a usable page in under ten seconds.
+
+This pitch will focus on what can be done to minimise the overall work done, to get to a usable page faster.
+
+
+## Recommendations
+
+In increasing level of trickiness to implement, options for improving page load time based on observations below:
+
+* Flag to skip reshape (~3.5s improvement)
+* Put settings into base page template (0.1s improvement)
+* Put tiles for lens details in base page template (0.4s+ improvement, final visual only)
+* Pre-prepare model server-side in form to be served (~2s improvement)
+* Change model format away from JSON and with better compression (0.5s+ improvement)
+
+
+## Results
+
+### Measurement
+
+Done using [Firefox Profiler] from Developer Tools.
+
+Sadly the profiler does not [support source maps] so to get useful function names out it's important to run against non-minified javascript. So, best to run multiple profiles, as the overall timings are better indications against the staging server, but need locally served javascript to get a good idea of where in the code time is being spent.
+
+There are also profilers for other browsers that could be used to compare results.
+
+Quality of network connection matters for the HTTP timings, and device matters for the browser and javascript timings.
+
+
+### Timings
+
+As an indicative timeline, the stages below form a linear sequence. Other things, such as fetching more JS files, CSS, and so on, happen in parallel off the critical path. Stitched together from remote and locally run profiles to get useful http and js time breakdowns.
+
+* 240ms HTTP GET html page (~1KB)
+* 140ms HTTP GET core js (12KB)
+* 95ms HTTP GET settings (~1KB)
+* 2400ms HTTP GET model (~2100KB gzip)
+  * 1950ms request and wait
+  * 435ms response
+* 230ms BROWSER `nsStreamLoader::OnStopRequest` (unicode and json parsing, gc)
+* ~3645ms JS `modelFromData`
+  * ~3615ms `reshapeModel`
+    * ~1700ms `addStakeholderOrder`
+    * ~820ms `_getSubjCache` (spread in various callsites)
+    * ~310ms `newView`
+    * ~290ms `labelLayersToLens`
+    * ~180ms `ontologyForCategoricalProperties`
+* ~800ms JS mostly react
+  * ~360ms `useFilteredItems`
+  * ~210ms (various map components and hooks)
+  * ~70ms `annotationDetails`
+  * ~45ms `annotationTriples`
+* 420ms HTTP GET blobstore tiles (~64KB in parallel)
+* ~70ms JS `hasRAGData`
+
+Time to get tiles from the blobstore can vary wildly - have seen up to 2000ms to complete - but they do load incrementally into the canvas under labels and the page is responsive by then.
+
+
+## Proposals in detail
+
+### No reshape
```

Comment:
I feel like this could absolutely come back to bite us when we apply our "no reshape" flag to something and then a year later it's all broken because we've updated the client and there's no longer the backwards compatibility provided by reshape. We need to be _really_ careful about this flag.